src/Detector/Golang/GolangDetector.cs

```diff
@@ -76,26 +77,23 @@ private string GetVersionFromGoDotMod(DetectorContext context)
                 // go 1.16
 
                 // Regex matching valid version format:
-                //      go 1
                 //      go 1.16
-                //      go 1.16.1
-                Regex regex = new Regex(@"^[\s]*go[\s]+[0-9]+(\.([0-9])+)?(\.([0-9])+)?[\s]*$");
+                //      go 1.16.7
+                Regex regex = new Regex(@"^[\s]*go[\s]+[0-9]+(\.([0-9])+)+[\s]*$");
                 foreach (var goDotModFileContentLine in goDotModFileContentLines)
                 {
-                    var goVersionLine = goDotModFileContentLine.Trim().Split(' ');
                     Match match = regex.Match(goDotModFileContentLine);
                     if (match.Success)
                     {
-                        return goVersionLine[1].Trim('\"').Trim('\'');
+                        return goDotModFileContentLine.Trim().Split(' ')[1].Trim('\"').Trim('\'');
```

Comment:
nit: add a description of what are your trimming and splitting

Reply:
Updated